### PR TITLE
[hjson-cpp] New port

### PR DESCRIPTION
**Pull request by wangtz0607**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/hjson/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "hjson",
+  "version": "2.4.1",
+  "port-version": 2,
```
Comment (Cheney-W) [inline ports/hjson/vcpkg.json:4]:
```suggestion
```
New ports should not have this entry.

File: versions/h-/hjson.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    {
+      "git-tree": "227ba93b152de42f53ca88379ec760f4b8265a59",
+      "version": "2.4.1",
+      "port-version": 2
+    },
+    {
+      "git-tree": "8a5381e9e9853306125764da6ac4860d6b55e233",
+      "version": "2.4.1",
+      "port-version": 1
+    },
```
Comment (Cheney-W) [inline versions/h-/hjson.json:12]:
```suggestion
```
There should be only one record here, and your local modifications should not be submitted to vcpkg.

**Review by Cheney-W — CHANGES_REQUESTED** [review]

**Comment by Cheney-W** [comment]
I think the name of this port should be `hjson-cpp` instead of `hjson`.
According to the https://repology.org/:
the upstream URL of `hjson-cpp` is https://github.com/hjson/hjson-cpp, consistent with what is used in `portfile.cmake`.
the upstream URL of `hjson` is https://github.com/hjson/hjson-py.

**Comment by wangtz0607** [comment]
I originally named the port `hjson-cpp`, but finally decided to use `hjson` to be consistent with existing naming conventions in vcpkg (e.g. `abseil` not `abseil-cpp`). I'm willing to make the change if needed.

**Comment by Cheney-W** [comment]
`abseil` is a port that was added to vcpkg early on, and at that time, the following rule did not exist:
```
The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
```
Later, in order to avoid impacting downstream usage, its name was not changed either.
However, the rule should be followed for all new ports added after it.

**Comment by wangtz0607** [comment]
Done.

**Review by Cheney-W — DISMISSED** [review]

**Comment by Cheney-W** [comment]
Usage test passed with below triplets:
```
x86-windows
x64-windows
x64-windows-static
```

File: ports/hjson-cpp/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    HEAD_REF master
+    PATCHES
+        fix-runtime-destination.patch
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        -DHJSON_ENABLE_INSTALL=ON
+)
```
Comment (JavierMatosD) [inline ports/hjson-cpp/portfile.cmake:15]: Should these be added as well?
```suggestion
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
    OPTIONS
        -DHJSON_ENABLE_INSTALL=ON
        -DHJSON_ENABLE_TEST=OFF
        -DHJSON_ENABLE_PERFTEST=OFF
        -DHJSON_VERSIONED_INSTALL=OFF
)
```

Comment (wangtz0607) [reply ports/hjson-cpp/portfile.cmake]: Done.

**Pull request merged**